ccl/multiregionccl: TestRegionAddDropEnclosingRegionalByRowOps failed

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3588667&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3588667&tab=artifacts#/) on master @ [0984f873c6170ab34afe6fee4661fc5f76ac0dee](https://github.com/cockroachdb/cockroach/commits/0984f873c6170ab34afe6fee4661fc5f76ac0dee):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table#01
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestRegionAddDropEnclosingRegionalByRowOps PKG=./pkg/ccl/multiregionccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/multiregion arulajmani
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-10669

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4632895&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4632895&tab=artifacts#/) on master @ [5bc7664396123e61f3546d2482b38e4aa0878ed6](https://github.com/cockroachdb/cockroach/commits/5bc7664396123e61f3546d2482b38e4aa0878ed6):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/e069c319bb13cd862ea8df887bc01cfc/logTestRegionAddDropEnclosingRegionalByRowOps1817312793
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegionAddDropEnclosingRegionalByRowOps
    region_test.go:323: -- test log scope end --
--- FAIL: TestRegionAddDropEnclosingRegionalByRowOps (157.09s)
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table
    region_test.go:289: 
        	Error Trace:	region_test.go:289
        	Error:      	Received unexpected error:
        	            	pq: error resolving REGIONAL BY ROW column type: type with ID 105 does not exist
        	Test:       	TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table
    --- FAIL: TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table (35.53s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4659257&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4659257&tab=artifacts#/) on master @ [8b367174769c89c0fcfe50986ed68d4650be7750](https://github.com/cockroachdb/cockroach/commits/8b367174769c89c0fcfe50986ed68d4650be7750):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/e069c319bb13cd862ea8df887bc01cfc/logTestRegionAddDropEnclosingRegionalByRowOps2656704589
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegionAddDropEnclosingRegionalByRowOps
    region_test.go:323: -- test log scope end --
--- FAIL: TestRegionAddDropEnclosingRegionalByRowOps (28.99s)
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table#01
    region_test.go:289: 
        	Error Trace:	region_test.go:289
        	Error:      	Received unexpected error:
        	            	pq: error resolving REGIONAL BY ROW column type: type with ID 105 does not exist
        	Test:       	TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table#01
    --- FAIL: TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table#01 (7.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4659259&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4659259&tab=artifacts#/) on master @ [8b367174769c89c0fcfe50986ed68d4650be7750](https://github.com/cockroachdb/cockroach/commits/8b367174769c89c0fcfe50986ed68d4650be7750):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/e069c319bb13cd862ea8df887bc01cfc/logTestRegionAddDropEnclosingRegionalByRowOps4099487117
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegionAddDropEnclosingRegionalByRowOps
    region_test.go:323: -- test log scope end --
--- FAIL: TestRegionAddDropEnclosingRegionalByRowOps (186.83s)
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table#01
    region_test.go:289: 
        	Error Trace:	region_test.go:289
        	Error:      	Received unexpected error:
        	            	pq: error resolving REGIONAL BY ROW column type: type with ID 105 does not exist
        	Test:       	TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table#01
    --- FAIL: TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table#01 (40.77s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4713913&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4713913&tab=artifacts#/) on master @ [29716850b181718594663889ddb5f479fef7a305](https://github.com/cockroachdb/cockroach/commits/29716850b181718594663889ddb5f479fef7a305):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/e069c319bb13cd862ea8df887bc01cfc/logTestRegionAddDropEnclosingRegionalByRowOps2726493038
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegionAddDropEnclosingRegionalByRowOps
    region_test.go:323: -- test log scope end --
--- FAIL: TestRegionAddDropEnclosingRegionalByRowOps (200.23s)
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table
    region_test.go:289: 
        	Error Trace:	region_test.go:289
        	Error:      	Received unexpected error:
        	            	pq: error resolving REGIONAL BY ROW column type: type with ID 105 does not exist
        	Test:       	TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table
    --- FAIL: TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table (44.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4766858&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4766858&tab=artifacts#/) on master @ [2ecd90c42c9460249385d6ec50360c04761ef4eb](https://github.com/cockroachdb/cockroach/commits/2ecd90c42c9460249385d6ec50360c04761ef4eb):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/e069c319bb13cd862ea8df887bc01cfc/logTestRegionAddDropEnclosingRegionalByRowOps2202890818
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegionAddDropEnclosingRegionalByRowOps
    region_test.go:323: -- test log scope end --
--- FAIL: TestRegionAddDropEnclosingRegionalByRowOps (202.20s)
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table
    region_test.go:311: 
        	Error Trace:	region_test.go:311
        	Error:      	Received unexpected error:
        	            	pq: relation "db.public.rbr" already exists
        	Test:       	TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table
    region_test.go:302: expected error boom, found dial tcp 127.0.0.1:37929: connect: connection refused
    --- FAIL: TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table (45.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4771794&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4771794&tab=artifacts#/) on master @ [71e32a6a5369dabde22043ca6ebbd11651404e5d](https://github.com/cockroachdb/cockroach/commits/71e32a6a5369dabde22043ca6ebbd11651404e5d):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/e069c319bb13cd862ea8df887bc01cfc/logTestRegionAddDropEnclosingRegionalByRowOps4272634044
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegionAddDropEnclosingRegionalByRowOps
    region_test.go:323: -- test log scope end --
--- FAIL: TestRegionAddDropEnclosingRegionalByRowOps (146.95s)
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table
    region_test.go:289: 
        	Error Trace:	region_test.go:289
        	Error:      	Received unexpected error:
        	            	pq: error resolving REGIONAL BY ROW column type: type with ID 105 does not exist
        	Test:       	TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table
    --- FAIL: TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table (32.99s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4820639&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4820639&tab=artifacts#/) on master @ [9bff1de7115984957978416d829feb582eb2664e](https://github.com/cockroachdb/cockroach/commits/9bff1de7115984957978416d829feb582eb2664e):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/e069c319bb13cd862ea8df887bc01cfc/logTestRegionAddDropEnclosingRegionalByRowOps2088400423
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegionAddDropEnclosingRegionalByRowOps
    region_test.go:323: -- test log scope end --
--- FAIL: TestRegionAddDropEnclosingRegionalByRowOps (188.72s)
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table
    region_test.go:289: 
        	Error Trace:	region_test.go:289
        	Error:      	Received unexpected error:
        	            	pq: error resolving REGIONAL BY ROW column type: type with ID 105 does not exist
        	Test:       	TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table
    --- FAIL: TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table (44.58s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4898179&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4898179&tab=artifacts#/) on master @ [b99291249d249ede65e732f89d3b3df8bf5472a4](https://github.com/cockroachdb/cockroach/commits/b99291249d249ede65e732f89d3b3df8bf5472a4):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/e069c319bb13cd862ea8df887bc01cfc/logTestRegionAddDropEnclosingRegionalByRowOps4252428914
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegionAddDropEnclosingRegionalByRowOps
    region_test.go:323: -- test log scope end --
--- FAIL: TestRegionAddDropEnclosingRegionalByRowOps (200.86s)
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/drop-region-succeed-create-rbr-table
    region_test.go:289: 
        	Error Trace:	region_test.go:289
        	Error:      	Received unexpected error:
        	            	pq: error resolving REGIONAL BY ROW column type: type with ID 105 does not exist
        	Test:       	TestRegionAddDropEnclosingRegionalByRowOps/drop-region-succeed-create-rbr-table
    --- FAIL: TestRegionAddDropEnclosingRegionalByRowOps/drop-region-succeed-create-rbr-table (46.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4907788&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4907788&tab=artifacts#/) on master @ [771432d1099e516dbc11827c5458886c176e73e3](https://github.com/cockroachdb/cockroach/commits/771432d1099e516dbc11827c5458886c176e73e3):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/e069c319bb13cd862ea8df887bc01cfc/logTestRegionAddDropEnclosingRegionalByRowOps1648151497
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegionAddDropEnclosingRegionalByRowOps
    region_test.go:323: -- test log scope end --
--- FAIL: TestRegionAddDropEnclosingRegionalByRowOps (186.42s)
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table
    region_test.go:289: 
        	Error Trace:	region_test.go:289
        	Error:      	Received unexpected error:
        	            	pq: error resolving REGIONAL BY ROW column type: type with ID 105 does not exist
        	Test:       	TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table
    --- FAIL: TestRegionAddDropEnclosingRegionalByRowOps/add-region-create-rbr-table (39.63s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4935360&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4935360&tab=artifacts#/) on master @ [7ea67fd5d4966508877fc88ba21bdd498ff162e8](https://github.com/cockroachdb/cockroach/commits/7ea67fd5d4966508877fc88ba21bdd498ff162e8):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/e069c319bb13cd862ea8df887bc01cfc/logTestRegionAddDropEnclosingRegionalByRowOps2516135976
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegionAddDropEnclosingRegionalByRowOps
    region_test.go:323: -- test log scope end --
--- FAIL: TestRegionAddDropEnclosingRegionalByRowOps (180.29s)
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table
    region_test.go:289: 
        	Error Trace:	region_test.go:289
        	Error:      	Received unexpected error:
        	            	pq: error resolving REGIONAL BY ROW column type: type with ID 105 does not exist
        	Test:       	TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table
    --- FAIL: TestRegionAddDropEnclosingRegionalByRowOps/drop-region-fail-create-rbr-table (39.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/multiregionccl.TestRegionAddDropEnclosingRegionalByRowOps [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5019404&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5019404&tab=artifacts#/) on master @ [2fdc0a62df08829dd96901fffb5fb1fbb0b0997d](https://github.com/cockroachdb/cockroach/commits/2fdc0a62df08829dd96901fffb5fb1fbb0b0997d):


```
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/e069c319bb13cd862ea8df887bc01cfc/logTestRegionAddDropEnclosingRegionalByRowOps1548660440
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegionAddDropEnclosingRegionalByRowOps
    region_test.go:323: -- test log scope end --
--- FAIL: TestRegionAddDropEnclosingRegionalByRowOps (191.96s)
=== RUN   TestRegionAddDropEnclosingRegionalByRowOps/drop-region-succeed-create-rbr-table
    region_test.go:289: 
        	Error Trace:	region_test.go:289
        	Error:      	Received unexpected error:
        	            	pq: error resolving REGIONAL BY ROW column type: type with ID 105 does not exist
        	Test:       	TestRegionAddDropEnclosingRegionalByRowOps/drop-region-succeed-create-rbr-table
    --- FAIL: TestRegionAddDropEnclosingRegionalByRowOps/drop-region-succeed-create-rbr-table (44.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionAddDropEnclosingRegionalByRowOps.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/create_table.go
+++ b/pkg/sql/create_table.go
@@ -251,6 +251,16 @@ func (n *createTableNode) startExec(params runParams) error {
 
 	colsWithPrimaryKeyConstraint := make(map[tree.Name]bool)
 
+	// Copy column definition slice, since we will modify it below. This
+	// ensures, that  nothing bad happens on a transaction retry errors, for
+	// example we implicitly add columns for REGIONAL BY ROW.
+	// Note: This is only a shallow copy and meant to deal with addition /
+	// deletion into the slice.
+	defsCopy := make(tree.TableDefs, 0, len(n.n.Defs))
+	defsCopy = append(defsCopy, n.n.Defs...)
+	defer func(originalDefs tree.TableDefs) { n.n.Defs = originalDefs }(n.n.Defs)
+	n.n.Defs = defsCopy
+
 	for _, def := range n.n.Defs {
 		switch v := def.(type) {
 		case *tree.UniqueConstraintTableDef:

